### ISSUE TITLE
(🐞) platform specified in inline configuration results in wacky behavior

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#12926]
It seems like even though this is not a per module option, it is still being applied to the module, but it's using the global(linux) type defs.
```py
import sys, winreg

if sys.platform == "win32":
    winreg.QueryValueEx

# Success: no issues found in 1 source file
```

```py
# mypy: platform=win32
import sys, winreg

if sys.platform == "win32":
    winreg.QueryValueEx  # test.py:5:5: error: Module has no attribute "QueryValueEx"  [attr-defined]
```
[playground](https://mypy-play.net/?mypy=latest&python=3.10&gist=0399afd74118d58296006d4470fb93e1)

- Relates to: #12342
